app/services/ping.rb

```diff
@@ -0,0 +1,85 @@
+class Ping
+  class << self
+    def ping
+      {
+        'foreman': {
+          database: ping_database
+        }
+      }.merge(plugins_ping)
+    end
+
+    def statuses
+      {
+        'foreman': {
+          version: SETTINGS[:version].full,
+          api: {
+            version: Apipie.configuration.default_version
+          },
+          plugins: Foreman::Plugin.all,
+          smart_proxies: statuses_smart_proxies,
+          compute_resources: statuses_compute_resources
+        }
+      }.merge(plugins_statuses).merge(ping) do |_key, old_val, new_val|
+        old_val.merge(new_val)
+      end
+    end
+
+    private
+
+    def ping_database
+      ActiveRecord::Base.connection.active?
+    end
+
+    def statuses_compute_resources
+      results = []
+      ComputeResource.all.index.map do |resource|
+        errors = resource.ping
+        results << {
+          name: resource.name,
+          status: errors.empty? ? 'ok' : 'FAIL',
```

Comment:
Are all ping implementations supposed to return `ok` and `FAIL`? Maybe put them in a constant?

Reply:
I'm using strings proposed by Katello plugin: https://github.com/Katello/katello/blob/bb87df839ef2e67b57b84d837f7063116809128b/app/models/katello/ping.rb#L3-L4. Since they were used in previous version of ping response, I've decided to leave them without changes and use in Foreman (although I wanted to see them in the same case; in upcase desirably).

Reply:
We probably can introduce the very same constants here in core and refactor Katello to use them.